### PR TITLE
add helpers for managing conditions

### PR DESCRIPTION
**Pull request by deads2k**
adds basic find/set/remove/status for conditions

/priority important-soon
@kubernetes/sig-api-machinery-pr-reviews 

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetStatusCondition sets the corresponding condition in conditions to newCondition.
+func SetStatusCondition(conditions *[]metav1.Condition, newCondition metav1.Condition) {
+	if conditions == nil {
+		conditions = &[]metav1.Condition{}
+	}
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go:13]: if nil is passed in, isn't the effect of this function lost?

File: staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package meta
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetStatusCondition sets the corresponding condition in conditions to newCondition.
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go:9]: document that `conditions` must be non-nil, and document what happens in the following cases:
1. if the condition of the specified type already exists (all fields of the existing condition are updated to `newCondition`, LastTransitionTime is set to now if the new status differs from the old status)
2. if a condition of the specified type does not exist (LastTransitionTime is set to now() if unset, and newCondition is appended)

File: staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+	existingCondition.Reason = newCondition.Reason
+	existingCondition.Message = newCondition.Message
+}
+
+// RemoveStatusCondition removes the corresponding conditionType from conditions.
+func RemoveStatusCondition(conditions *[]metav1.Condition, conditionType string) {
+	if conditions == nil {
+		conditions = &[]metav1.Condition{}
+	}
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go:40]: same question, if nil, can't we just return early?

File: staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	newConditions := []metav1.Condition{}
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go:41]:
```suggestion
	newConditions := make([]metav1.Condition, 0, len(conditions)-1)
```

**Comment by liggitt** [comment]
a couple nits and doc requests, mostly lgtm

**Comment by liggitt** [comment]
/approve 

nit on boilerplate header, lgtm otherwise

**Comment by deads2k** [comment]
verify fixed.  Tagging per comment.

**Pull request merged**